### ISSUE TITLE
Attempted to call an undefined method named "fetchColumn" viewer audit

### ISSUE THREAD

**Issue by tomekkowalczyk** [DamienHarper/auditor-bundle#280]
| Q                           | A
| ----------------------------| -----------------------
| `auditor-bundle` version    | 1.3.0
| PHP version                 | 8.0.9
| Database                    | MySQL 

#### Summary

Address URL: https://app.localhost/audit

**BUG info:** 

`Attempted to call an undefined method named "fetchColumn" of class "Doctrine\DBAL\Result".`

doctrine/dbal 3.1.3
Symfony 5.3.9

**Comment by DamienHarper** [comment]
@tomekkowalczyk `doctrine/dbal` 3.x is not supported by `auditor` 1.3.0 but unfortunately, 1.3.0 didn't prevent to install/update to `doctrine/dbal` 3.x (see #279)

It has been fixed in `auditor` 1.3.1 which prevents installation of `doctrine/dbal` 3.x

You should downgrade `doctrine/dbal` to its latest 2.x version then update `auditor` to 1.3.1.

**Comment by tomekkowalczyk** [comment]
@DamienHarper thanks for the answer

  - Upgrading damienharper/auditor (1.3.0 => 1.3.1)
  - Downgrading doctrine/dbal (3.1.3 => 2.13.4)

looks like it's ok

**Comment by seizan8** [comment]
> @tomekkowalczyk `doctrine/dbal` 3.x is not supported by `auditor` 1.3.0 but unfortunately, 1.3.0 didn't prevent to install/update to `doctrine/dbal` 3.x (see #279)
> 
> It has been fixed in `auditor` 1.3.1 which prevents installation of `doctrine/dbal` 3.x
> 
> You should downgrade `doctrine/dbal` to its latest 2.x version then update `auditor` to 1.3.1.

Is there not a besser fix to this?
I think the problem appears in `DH\Auditor\Provider\Doctrine\Persistence\Reader\Reader` on line 50 
```
$query = new Query($this->getEntityAuditTableName($entity), $storageService->getEntityManager()->getConnection());
```
getConnection returns a Connection from `Orm\DBAL` namespace. When `createQueryBuilder()` is later called on the connection it returns `Doctrine\DBAL\Query\QueryBuilder`.
However, `Doctrine\ORM\QueryBuilder` should have been used.

`DH\Auditor\Provider\Doctrine\Persistence\Reader\Query` only uses the connection to get the QueryBuilder. So, why is it even passed to it?

Passing the EntityManager to the Query and call `createQueryBuilder` on it instead of the connection should solve the issue and not require a lower DBAL version. 
```
$query = new Query($this->getEntityAuditTableName($entity), $storageService->getEntityManager());
```
